stdlib/public/core/Collection.swift

```diff
@@ -398,6 +392,15 @@ extension Collection {
   }
 }
 
+/// Supply a default `next()` method for `Collection` models that
+/// use some model of `Strideable` as their `Index`.
```

Comment:
You can also improve `advance()` and `distance()`.  If you don't want to implement this now, please just add a `FIXME: swift-3-indexing-model` tag.